antarest/study/storage/variantstudy/model/command/update_binding_constraint.py

```diff
@@ -72,37 +58,48 @@ def _update_matrices_names(
         )
     elif existing_operator == new_operator:
         return  # nothing to do
-    elif existing_operator != BindingConstraintOperator.BOTH and new_operator != BindingConstraintOperator.BOTH:
-        matrix_node = parent_folder_node.get_node([f"{binding_constraint_id}_{ALIAS_OPERATOR_MAP[existing_operator]}"])
-        assert isinstance(
-            matrix_node, LazyNode
-        ), f"Node type not handled yet: LazyNode expected, got {type(matrix_node)}"
-        new_matrix_node = parent_folder_node.get_node([f"{binding_constraint_id}_{ALIAS_OPERATOR_MAP[new_operator]}"])
-        assert isinstance(
-            new_matrix_node, LazyNode
-        ), f"Node type not handled yet: LazyNode expected, got {type(new_matrix_node)}"
-        matrix_node.rename_file(new_matrix_node)
+
+    operator_matrices_map = {
+        BindingConstraintOperator.EQUAL: "eq",
+        BindingConstraintOperator.GREATER: "gt",
+        BindingConstraintOperator.LESS: "lt",
+    }
+
+    def get_matrix_path(constraint_folder: Path, matrix_id: str) -> Path:
+        raw_file = constraint_folder / f"{matrix_id}.txt"
+        return raw_file if raw_file.exists() else raw_file.with_suffix(".txt.link")
+
+    def get_target_path(existing_path: Path, constraint_folder: Path, matrix_id: str) -> Path:
```

Comment:
It was better to not manipulate directly files in this layer of the code (commands implementation): the implementation details of handling links or actual files should be left in the file tree code. Possibly, we'll be able to re-use copy and rename methods elsewhere in the future.

Reply:
Agreed but with this dev some nodes won't exist so these methods won't work anymore.